dali/operators/reader/loader/video/video_test_base.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_LOADER_VIDEO_VIDEO_TEST_BASE_H_
+#define DALI_OPERATORS_READER_LOADER_VIDEO_VIDEO_TEST_BASE_H_
+
+#include <gtest/gtest.h>
+#include <opencv2/core.hpp>
+
+
+namespace dali {
+class VideoTestBase : public ::testing::Test {
+ public:
+  const int NumVideos() const { return cfr_frames_.size(); }
+
+  const int NumFrames(int i) const { return cfr_frames_[i].size(); }
+
+  const int Channels() const { return 3; }
+
+  const int Width(int i) const { return cfr_frames_[i][0].cols; }
+
+  const int Height(int i) const { return cfr_frames_[i][0].rows; }
+
+  const int FrameSize(int i) const { return Height(i) * Width(i) * Channels(); }
+
+  void ComapreFrames(const uint8_t *frame, const uint8_t *gt, int size, int eps = 0);
+
+  uint8_t *GetCfrFrame(int video_id, int frame_id) { return cfr_frames_[video_id][frame_id].data; }
```

Comment:
In general, OpenCV Matrices are not guaranteed to be continuous. It's empirical knowledge that images loaded with `imread` are indeed continuous, but the documentation doesn't make any assumptions about it.

Reply:
I think that for the sake of simplicity you can just assert continuity after loading the reference images - after all, it's just a test.

Reply:
Done, added check for `isContinuous()` after the load.
I looked a bit and as far as I can tell, OpenCV guarantees continuity for all matrices created with `create` and `imread` uses that. But there is no plain guarantee that `imread` will always do this.